### PR TITLE
vnic dlpar with hmc cli

### PR DESCRIPTION
**Pull request by vaishnavibhat**
vnic dlpar with hmc cli commands. Existing dlapr
test is with drmgr commands only. Adding a new test
for testing dlpar operation with hmc cli way.

Signed-off-by: Vaishnavi Bhat <vaishnavi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by vaishnavibhat** [comment]
avocado run --test-runner runner  network_virtualization.py:NetworkVirtualization.test_vnic_hmc_dlpar -m network_virtualization.yaml 
JOB ID     : 80b1f4bd8cb3be239c268a4e88c86bad8caa1c7d
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2022-05-20T07.46-80b1f4b/job.log
 (1/1) network_virtualization.py:NetworkVirtualization.test_vnic_hmc_dlpar;run-a172: PASS (238.83 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2022-05-20T07.46-80b1f4b/results.html
JOB TIME   : 255.91 s

[job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/8751648/job.log)

File: io/net/virt-net/network_virtualization.py [hunk 1]
```diff
@@ -523,6 +523,76 @@ def test_vnic_dlpar(self):
...
+        Lists the count of backing devices
+        '''
+        count = 0
+        output = self.backing_dev_list()
+        for i in output.splitlines():
+            if i.startswith('%s,' % slot):
+                count = len(i.split(',')[1:])
+        return count
+
+    def wait_interface(self, device_name):
```
Comment (abdhaleegit) [inline io/net/virt-net/network_virtualization.py:584]: Just thinking can we push wait_interface to utils ?

File: io/net/virt-net/network_virtualization.py [hunk 2]
```diff
@@ -523,6 +523,76 @@ def test_vnic_dlpar(self):
...
+                    self.fail("Unable to bring up the link on the Network virtualized device")
+
+                time.sleep(5)
+
+                if networkinterface.ping_check(peer_ip, count=5) is not None:
+                    self.fail("dlpar has affected Network connectivity")
+                count += 1
+        self.check_dmesg_error()
+
+    def backing_dev_count_w_slot_num(self, slot):
```
Comment (abdhaleegit) [inline io/net/virt-net/network_virtualization.py:573]: also all the vnic functions to network utils

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

File: io/net/virt-net/network_virtualization.py [hunk 3]
```diff
@@ -523,6 +523,76 @@ def test_vnic_dlpar(self):
...
+        '''
+        Lists the count of backing devices
+        '''
```
Comment (Naresh-ibm) [inline io/net/virt-net/network_virtualization.py:576]: Please format this docstring using the PEP-257 style.

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

**Comment by abdhaleegit** [comment]
@vaishnavibhat Please look into the comments, and also rebase your code as there is merge conflict

**Comment by abdhaleegit** [comment]
@vaishnavibhat please look into this

File: io/net/virt-net/network_virtualization.py [hunk 4]
```diff
@@ -523,6 +523,76 @@ def test_vnic_dlpar(self):
...
+        for slot_no, device_ip, netmask, mac, peer_ip, sriov_port, adapter_id in zip(self.slot_num,
+                                                                                     self.device_ip,
+                                                                                     self.netmask,
+                                                                                     self.mac_id,
+                                                                                     self.peer_ip,
+                                                                                     self.sriov_port,
+                                                                                     self.backing_adapter_id):
+            self.update_backing_devices(slot_no)
+            device_name = self.find_device(mac)
+            device = self.find_device(mac)
```
[the remark this answers is not in the document]
Comment (vaishnavibhat) [reply io/net/virt-net/network_virtualization.py]: Taken care and resent the patch. 
Thank you.

File: io/net/virt-net/network_virtualization.py [hunk 5]
```diff
@@ -528,10 +523,79 @@ def test_vnic_dlpar(self):
...
+
+                if networkinterface.ping_check(peer_ip, count=5) is not None:
+                    self.fail("dlpar has affected Network connectivity")
+                count += 1
+        self.check_dmesg_error()
+
+    def backing_dev_count_w_slot_num(self, slot):
+        """
+        Lists the count of backing devices
+        """
```
Comment (Naresh-ibm) [inline io/net/virt-net/network_virtualization.py:575]: @vaishnavibhat this has not yet taken care.
let me know if you need any help here

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

**Review by abdhaleegit — APPROVED** [review]
LGTM

File: io/net/virt-net/network_virtualization.py [hunk 6]
```diff
@@ -528,10 +523,84 @@ def test_vnic_dlpar(self):
...
+                count = len(i.split(',')[1:])
+        return count
+
+    def wait_interface(self, device_name):
+        """
+        Wait till interface comes up
+        param device_name: vnic device that is tested
+
+        returns: if the device is up or down
+        """
```
Comment (Naresh-ibm) [inline io/net/virt-net/network_virtualization.py:591]: @vaishnavibhat Please use the following reference for proper pep standards,

```
def get_dir_mountpoint(dir_path):
    """
    get mounted disk name that is mounted on given dir_path

    :param dir_path: absolute directory path
    :type dir_path: str

    :returns: returns disk name which mounted on given dir_path
    :rtype: str
    """

```

File: io/net/virt-net/network_virtualization.py [hunk 7]
```diff
@@ -528,10 +523,89 @@ def test_vnic_dlpar(self):
...
+        self.check_dmesg_error()
+
+    def backing_dev_count_w_slot_num(self, slot):
+        """
+        Lists the count of backing devices
+        :param slot: vnic slot
+        :type slot: str
+
+        :returns backing device count
+        :rtype int
```
Comment (Naresh-ibm) [inline io/net/virt-net/network_virtualization.py:579]: @vaishnavibhat 
slight modifications.
this should be as below.
```
:returns:  backing device count
:rtype:  int
```

File: io/net/virt-net/network_virtualization.py [hunk 8]
```diff
@@ -528,10 +523,89 @@ def test_vnic_dlpar(self):
...
+                count = len(i.split(',')[1:])
+        return count
+
+    def wait_interface(self, device_name):
+        """
+        Wait till interface comes up
+        :param device_name: vnic device that is tested
+        :type device_name: str
+
+        :returns if the device is up or down
```
Comment (Naresh-ibm) [inline io/net/virt-net/network_virtualization.py:594]: I feel this message can be modified slightly as below along with the above suggested changes for docstring,

:returns:  returns True if device is up else False

File: io/net/virt-net/network_virtualization.py [hunk 9]
```diff
@@ -528,10 +523,89 @@ def test_vnic_dlpar(self):
...
+        count = 0
+        output = self.backing_dev_list()
+        for i in output.splitlines():
+            if i.startswith('%s,' % slot):
+                count = len(i.split(',')[1:])
+        return count
+
+    def wait_interface(self, device_name):
+        """
+        Wait till interface comes up
```
Comment (Naresh-ibm) [inline io/net/virt-net/network_virtualization.py:590]: Leave a line gap after this

File: io/net/virt-net/network_virtualization.py [hunk 10]
```diff
@@ -528,10 +523,89 @@ def test_vnic_dlpar(self):
...
+                time.sleep(5)
+
+                if networkinterface.ping_check(peer_ip, count=5) is not None:
+                    self.fail("dlpar has affected Network connectivity")
+                count += 1
+        self.check_dmesg_error()
+
+    def backing_dev_count_w_slot_num(self, slot):
+        """
+        Lists the count of backing devices
```
Comment (Naresh-ibm) [inline io/net/virt-net/network_virtualization.py:574]: a line gap after this

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

File: io/net/virt-net/network_virtualization.py (hunk 6 above)
Comment (vaishnavibhat) [reply io/net/virt-net/network_virtualization.py:596]: Resent patch with the above format.

**Pull request merged**